include/solvers/petsc_linear_solver.h

```diff
@@ -45,6 +45,24 @@
 // Give them an obscure name to avoid namespace pollution.
 extern "C"
 {
+#if PETSC_RELEASE_LESS_THAN(3,0,1)
```

Comment:
This check should be `PETSC_VERSION_LESS_THAN`. 

The `PETSC_RELEASE_LESS_THAN` macro is used when you need to check *both* a minimum version *and* that you are using an actual released version of PETSc (as opposed to some branch like master, next, etc.). It's usually used with negation (`!PETSC_RELEASE_LESS_THAN`) to compile with code that is currently only in PETSc master.

I see a few other places where the macros are wrong in this file, so I can fix that in a separate PR.

Reply:
👍